### PR TITLE
Implemented mock classes 

### PR DESCRIPTION
**Pull request by lbrlekovi20**
Implemented mock classes for mock data and the ability to switch between using mock data and actual database data using appsettings.json. Now the REST service funcitonalities are fully testable.

### CODE REVIEW

File: RESTservice-API/Data/MockPositionHistoryRepository.cs [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+namespace RESTservice_API.Data
+{
+    public class MockPositionHistoryRepository : IPositionHistoryRepository
+    {
+        private List<PositionHistory> _positionHistories;
+
+        public MockPositionHistoryRepository()
+        {
+            // Initialize some mock data
```
Comment (vdraganic21) [inline RESTservice-API/Data/MockPositionHistoryRepository.cs]: Remove comment.

File: RESTservice-API/appsettings.json [hunk 2]
```diff
@@ -6,6 +6,7 @@
     }
   },
   "AllowedHosts": "*",
+  "UseMockData": true, // Set to true for mock data, false for real database
```
Comment (vdraganic21) [inline RESTservice-API/appsettings.json:9]: A nice way to quickly change between mock mode and actual database usage. 👍 
Be careful to not overcomplicate things with appsettings in the future.

File: RESTservice-API/Program.cs [hunk 3]
```diff
@@ -1,25 +1,32 @@
 using Microsoft.EntityFrameworkCore;
 using RESTservice_API.Data;
-using RESTservice_API.Models;
 
 var builder = WebApplication.CreateBuilder(args);
 
-// Add DbContext
-builder.Services.AddDbContext<ApplicationDbContext>(options =>
-options.UseSqlServer(builder.Configuration.GetConnectionString("DefaultConnection")));
+// Register the services based on whether to use mock data or the real database
```
Comment (vdraganic21) [inline RESTservice-API/Program.cs]: Remove comments.

File: RESTservice-API/Program.cs [hunk 4]
```diff
@@ -1,25 +1,32 @@
...
+if (useMockData)
+{
+    // Register mock repositories
+    builder.Services.AddSingleton<IAssetRepository, MockAssetRepository>();
+    builder.Services.AddSingleton<IPositionHistoryRepository, MockPositionHistoryRepository>();
+}
+else
+{
+    // Register the real database context and repository
+    builder.Services.AddDbContext<ApplicationDbContext>(options =>
```
Comment (vdraganic21) [inline RESTservice-API/Program.cs:15]: Consider refactoring these blocks of code into functions.

File: RESTservice-API/Interfaces/PositionHistoryRepository.cs [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+        private readonly ApplicationDbContext _context;
+
+        public PositionHistoryRepository(ApplicationDbContext context)
+        {
+            _context = context;
+        }
+
+        public IEnumerable<PositionHistory> GetAllPositionHistories()
+        {
+            // Ensure that this method returns IEnumerable<PositionHistory>
```
Comment (vdraganic21) [inline RESTservice-API/Interfaces/PositionHistoryRepository.cs]: Remove comments.

File: RESTservice-API/Controllers/AssetsController.cs
Comment (vdraganic21) [inline RESTservice-API/Controllers/AssetsController.cs:1]: Controller now uses repository for data access. Seems fine.

**Review by vdraganic21 — CHANGES_REQUESTED** [review]
Implementation seems fine. Program runs.
I requested some changes to be made, please consider them.
Keep up the good work with other repositories.
Make the changes and I'll approve the PR! 😃

**Review by vdraganic21 — APPROVED** [review]
lgtm

**Review by aslavicek21 — APPROVED** [review]
Great work; approved.

**Pull request merged**